deploy/deploy_arbitrum_lpt_mocks.ts

```diff
@@ -0,0 +1,31 @@
+import {HardhatRuntimeEnvironment} from "hardhat/types"
+import {DeployFunction} from "hardhat-deploy/types"
+import {contractId} from "../utils/helpers"
+
+const func: DeployFunction = async function(hre: HardhatRuntimeEnvironment) {
+    const {deployments, getNamedAccounts, ethers} = hre
+    const {deploy, get} = deployments
+
+    const {deployer} = await getNamedAccounts()
+
+    const deployResult = await deploy("L2LPTDataCacheMock", {
+        from: deployer,
+        log: true
+    })
+
+    const controllerDeployment = await get("Controller")
+    const controller = await ethers.getContractAt(
+        "Controller",
+        controllerDeployment.address
+    )
+
+    const id = contractId("L2LPTDataCache")
+    await controller.setContractInfo(
+        id,
+        deployResult.address,
+        "0x1111111111111111111111111111111111111111"
+    )
+}
+
+func.tags = ["ARBITRUM_LPT_MOCK"]
```

Comment:
I wonder it `ARBITRUM_LPT_MOCKS` is explicit enough. The same with the file name `deploy_arbitrum_lpt_mocks.ts`. I think that, as per my previous comment, the best would be to not have this `L2LPTDataCacheMock` contract at all. But if we really do need it, then I'd consider renaming it something like `DummyMinter` or maybe `DummyL1Supply` and then follow the naming in the tag and this file name.

Reply:
> Why do we need to hardcode l1CirculatingSupply to 0 in the geth-with-protocol docker image? What would happen if we don't have it? Or if we deploy the real Minter contract, not the mock one? Maybe there is a way to avoid having this L2LPTDataCacheMock.sol?

We are deploying the actual Minter contract. The difference is that the 'post-confluence' Minter contract [relies on an external dependency to calculate the total supply of livepeer tokens](https://github.com/livepeer/protocol/blob/confluence/contracts/token/Minter.sol#L228). The actual is defined elsewhere - [here in the arbitrum-lpt repo](https://github.com/livepeer/arbitrum-lpt-bridge/blob/main/contracts/L2/gateway/L2LPTDataCache.sol#L98-L111)- and, as I understand, its responsibility is to ensure that the global supply takes into account the l1/mainnet total supply difference if necessary (i.e.: in the case the inflation on L1).
Since it seems that the agreed-upon direction was to[ keep the initial docker workflow fairly simple and not supporting cross-chain testing](https://github.com/livepeer/go-livepeer/issues/2297), I opted for the usage of a mock.

> I wonder it ARBITRUM_LPT_MOCKS is explicit enough. The same with the file name deploy_arbitrum_lpt_mocks.ts. I think that, as per my previous comment, the best would be to not have this L2LPTDataCacheMock contract at all. But if we really do need it, then I'd consider renaming it something like DummyMinter or maybe DummyL1Supply and then follow the naming in the tag and this file name.

IMHO I think it falls under the definition of a mock, but if you insist on it I can change the name.

Reply:
Thanks for the explanation. It makes sense.

About the naming, since we already use `Dummy` somewhere in the code, It'd use Dummy for it. What you have is closer to Dummy or Stub than to a Mock. Mocking is more about [recording behavior and setting the expected values.](https://stackoverflow.com/questions/346372/whats-the-difference-between-faking-mocking-and-stubbing).